### ISSUE TITLE
[Feature] report on balance

### ISSUE THREAD

**Issue by vincentDcmps** [actualbudget/actual#3248]
### Verified feature request does not already exist?

- [X] I have searched and found no existing issue

### 💻

- [ ] Would you like to implement this feature?

### Pitch: what problem are you trying to solve?

I would like see evolution throught time of my balance f

### Describe your ideal solution to this problem

for example i put a part of my budget in a category each month to schedule an investment 
 I would like see in a report balance evolution. to determine in how many time I will get the goal

**Comment by pitch-patch** [comment]
Would love this!

Being able to see the Available balance across selected categories (even if they’re in different groups) would be super helpful. Especially for tracking savings or buffers.

**Comment by mdhowle** [comment]
I would like this as well. 

For instance, if I allocate $100 each month for Groceries.  In March, I spend $50 of the $100; in April $60, and May $70.  That would leave me with a surplus of $50, $40, and $30 respectively.

<table>
  <tr>
    <td>Category</td>
    <td>Budgeted</td>
    <td>Spent</td>
    <td>Balance</td>
    <td>Note</td>
  </tr>
  <tr>
    <td colspan="5">March</td>
  </tr>
  <tr><td>Groceries</td><td>$100</td><td>$50</td><td>$50</td></tr>
   <tr> <td colspan="5">April</td></td>
  <tr><td>Groceries</td><td>$100</td><td>$60</td><td>$90</td><td>$100 - $60 = $40 surplus plus the $50 from last month</td></tr>
   <tr> <td colspan="5">May</td></tr>
  <tr><td>Groceries</td><td>$100</td><td>$70</td><td>$120</td><td>$100 - $70 = $30 surplus plus the $90 from last month</td></tr>
</table>

I'd like to see the graph like below.  The red "Budget total" line showing the total category balance growing each month, which I believe this feature request is suggesting.

 In addition, I would like a difference between the monthly category balance and category payments as shown in the blue "Budget surplus" line. This line, using the example above, shows that my surplus is shrinking as I'm spending more on groceries monthly. This would be valuable insight.

<img width="606" height="341" alt="Image" src="https://github.com/user-attachments/assets/faba19f9-0656-4854-8c1e-94ddf7d44c72" />

**Comment by mgdesign** [comment]
I think that budget balance could be a useful addition to the reports also in other scenarios. 
For example, consider this situation where you need to budget per category but those categories are connected to a specific account. 

<img width="1102" height="717" alt="Image" src="https://github.com/user-attachments/assets/28654336-bbd1-439a-b0e1-a81b46b54430" />

You may want to split the categories in more groups for better reporting on the expenses (all monthly, only family annual, etc) but that makes more difficult to check that all the sum of the current budget balance of specific categories types corresponds to the balance of the dedicated account for those categories.

A report that exposes the current balance of a group of categories could solve this scenario: for example I could add a chart to show the budget balance of all the categories of the "family" type and quickly check that the balance of the account corresponds to the allocated budget, since that will be the account I will use for those expenses.

In other words, I could check not only that the budget is allocated, but also that the account I will use has the cash for that budget.

What do you think?

**Comment by david-kooi-dsi** [comment]
IMO, this is a nessecary addition. 

The main feature I need is to track actual vs budgeted differences. 
Given the past 6 months, I want to see how much each category overs/under-spent. 

This is a standard accounting metric.

At a different level, I've noticed the reports only deal with transactions and not categories....which if the spirit of Actual is envelope budgeting...you would think there would be a way to visualize the balance of each envelope.

**Comment by darrenwilliamwilson** [comment]
> I think that budget balance could be a useful addition to the reports also in other scenarios. For example, consider this situation where you need to budget per category but those categories are connected to a specific account.
> 
> <img alt="Image" width="1102" height="717" src="https://private-user-images.githubusercontent.com/5602741/532380762-28654336-bbd1-439a-b0e1-a81b46b54430.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3Njk5NDM5MTIsIm5iZiI6MTc2OTk0MzYxMiwicGF0aCI6Ii81NjAyNzQxLzUzMjM4MDc2Mi0yODY1NDMzNi1iYmQxLTQzOWEtYjBlMS1hODFiNDZiNTQ0MzAucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI2MDIwMSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNjAyMDFUMTEwMDEyWiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YzFjMzg0OTI4NjgwNzcwNGU0MDhmMDE3NGVkMDY0Nzc0MTAyMThlZWVlNjM4MzA1YzgxY2M3Yjk2Njk2ZmQ3NSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.yIBaKusI-m59E3dev8ZPJiiDa4Imy46Kf0sgoGtnj9k">
> You may want to split the categories in more groups for better reporting on the expenses (all monthly, only family annual, etc) but that makes more difficult to check that all the sum of the current budget balance of specific categories types corresponds to the balance of the dedicated account for those categories.
> 
> A report that exposes the current balance of a group of categories could solve this scenario: for example I could add a chart to show the budget balance of all the categories of the "family" type and quickly check that the balance of the account corresponds to the allocated budget, since that will be the account I will use for those expenses.
> 
> In other words, I could check not only that the budget is allocated, but also that the account I will use has the cash for that budget.
> 
> What do you think?

This exactly why I would like this feature. This is like YNAB's custom views method. Allows me to track certain categories to a certain spending account so I can quickly if that account balance is correct

**Comment by jacobjeffery** [comment]
I was just trying to create a report for my on hand savings categories which led me here. I would very much like this feature to be implemented!
